spring-boot-actuator/src/main/java/org/springframework/boot/actuate/security/AbstractAuthenticationAuditListener.java

```diff
@@ -32,4 +34,17 @@
 public abstract class AbstractAuthenticationAuditListener implements
 		ApplicationListener<AbstractAuthenticationEvent>, ApplicationEventPublisherAware {
 
+	private ApplicationEventPublisher publisher;
+
+	@Override
+	public void setApplicationEventPublisher(ApplicationEventPublisher publisher) {
+		this.publisher = publisher;
+	}
+
+	protected void publish(AuditEvent event) {
```

Comment:
There's not much point in this method being `protected` when there's no way to access the publisher. A getter for `publisher` seems appropriate.